File: baa.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>

int main(){
    int runs = 10000000, i = 0, j = 0, car, sel, away;
    float result;

    fprintf(stdout, "%un", (unsigned)time(NULL));
    printf("\n");

    for(i = 0; i < runs; i++){
    car = rand() % 3;
    away = car;
    sel = rand() % 3;

    if(car == sel){
        while(away == car){
            away = rand() % 3;
            sel = 3 - (sel + away);
        }
    }else{
        away = 3 - (car + sel);
        sel = 3 - (sel + away);
    }

    if (sel == car){
        j += 1;
    }
    }
    fprintf(stdout, "%un", (unsigned)time(NULL));
    printf("%d\n", j);
    result = (float) j / (float) runs;
    printf("%f\n", result);
}

File: baa.py
import random
import time

j = 0
runs = 1000000
stime = time.time()
for i in range(runs):
    car = random.randint(0, 2)
    away = car
    select = random.randint(0, 2)

    if car == select:
        while away == car:
            away = random.randint(0, 2)
            select = 3 - (select + away)

    else:
        away = 3 - (car + select)
        select = 3 - (select + away)

    if select == car:
        j += 1

print(time.time() - stime)
print(j/runs)
